client/my-sites/plugins/plugin-details-header/index.jsx

```diff
@@ -17,6 +17,16 @@ const PluginDetailsHeader = ( { plugin, isPlaceholder, isJetpackCloud } ) => {
 
 	const selectedSite = useSelector( getSelectedSite );
 
+	// This gets the version of the plugin on the last site or
+	// the wporg version if that doesn't exist. This is preserving
+	// some pre-existing behavior during a refactor and doesn't
+	// necessarily have to remain this way in the future.
+	let version = plugin?.version;
+	if ( plugin.sites ) {
+		const sites = Object.values( plugin.sites );
+		version = sites.findLast( ( site ) => site.version ).version;
+	}
+
```

Comment:
I wonder if we should extract this to a selector. It's rarely a good idea to do such calculations at render time.

Reply:
I don't want to put this in a selector because this was basically undefined behavior, and making it a reusable selector would make it official. We need to answer the question of what version a plugin should show here if it has multiple versions on multiple sites. Maybe it could be a range, or it could be the most recent version of the plugin which is installed, or it could be the most recent version of a plugin which is available. No matter the case this logic should not be used anymore. I have kept this logic here because I don't want these changes to alter what is shown in the UI in any way.

@vitozev might be able to clarify if this is something that Jetpack Genesis could take a deeper look at.

Reply:
We can definitely look into this. If you have more context (e.g., where precisely in the UI we show this, etc.), I would appreciate it if you add this to our [maintenance board](1202619025189113-asboard), and we'll triage it.

Reply:
I created 1202619025189113-as-1204688670518465